From fef0ff96f627e6f41278cc3f71e398785269555c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:18 -0400
Subject: [usb] ch341: only wake up MSR queue on changes

Message-id: <1410887463-20674-247-git-send-email-dzickus@redhat.com>
Patchwork-id: 95374
O-Subject: [RHEL7 PATCH 246/591] USB: ch341: only wake up MSR queue on changes
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d984fe91a85e49841449b120cda5a50fed1fb126
Author: Johan Hovold <jhovold@gmail.com>
Date:   Thu Jan 2 22:49:31 2014 +0100

    USB: ch341: only wake up MSR queue on changes

    Only wake up MSR wait queue on actual modem-status changes.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/ch341.c b/drivers/usb/serial/ch341.c
index 025b785..acc8865 100644
--- a/drivers/usb/serial/ch341.c
+++ b/drivers/usb/serial/ch341.c
@@ -460,6 +460,9 @@ static void ch341_update_line_status(struct usb_serial_port *port,
  if (data[1] & CH341_MULT_STAT)
   dev_dbg(&port->dev, "%s - multiple status change\n", __func__);
 
+ if (!delta)
+  return;
+
  if (delta & CH341_BIT_DCD) {
   tty = tty_port_tty_get(&port->port);
   if (tty) {
-- 
1.7.1